go/runtime/host/protocol/connection.go

```diff
@@ -167,10 +203,25 @@ func (c *connection) Call(ctx context.Context, body *Body) (*Body, error) {
 		return nil, ErrNotReady
 	}
 
-	return c.call(ctx, body)
+	b, err := c.call(ctx, body)
+	if viper.GetString(metrics.CfgMetricsMode) != metrics.MetricsModeNone {
+		if err != nil {
```

Comment:
This will miss calls that use the internal `call` directly. Instead of doing it here to catch the error you can use named return values like so:
```golang
func (c *connection) call(ctx context.Context, body *Body) (result *Body, err error) {
```

And in the defer handler you then have the `err` available so you can check its value (e.g., is it non-nil) and update the corresponding counter. This way you also avoid having the metrics-enabled check twice.

Reply:
Good idea!

Reply:
:+1: